api/v1alpha2/azuremachine_types.go

```diff
@@ -30,7 +30,7 @@ const (
 
 // AzureMachineSpec defines the desired state of AzureMachine
 type AzureMachineSpec struct {
-	AvailabilityZone string `json:"availabilityZone"`
+	AvailabilityZone string `json:"availabilityZone,omitempty"`
```

Comment:
Is this required or optional?

Reply:
@vincepri -- I think leveraging AZs should be a goal as they provide a better guarantees than standard VMs.

The field isn't required, but the code prevents a foot-gun situation by choosing an AZ for you if you haven't already done so.

Reply:
How about then to have something like this

```suggestion
        // <Add description here>
        // +optional
	AvailabilityZone *string `json:"availabilityZone,omitempty"`
```

This helps us having the nil case (not specified) as being the default one, without supplying the empty string